PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['gamification.badge']
-    _inherit = ['gamification.challenge']
-    _inherit = ['res.partner']

Total Changes: 7

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/survey/models/badge.py
+++ b/addons/survey/models/badge.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class GamificationBadge(models.Model):
-    _inherit = ['gamification.badge']
+    _inherit = 'gamification.badge'
 
     survey_ids = fields.One2many('survey.survey', 'certification_badge_id', 'Survey Ids')
     survey_id = fields.Many2one('survey.survey', 'Survey', compute='_compute_survey_id', store=True)

--- a/addons/survey/models/challenge.py
+++ b/addons/survey/models/challenge.py
@@ -5,7 +5,7 @@ from odoo import models, fields
 
 
 class GamificationChallenge(models.Model):
-    _inherit = ['gamification.challenge']
+    _inherit = 'gamification.challenge'
 
     challenge_category = fields.Selection(selection_add=[
         ('certification', 'Certifications')

--- a/addons/survey/models/res_partner.py
+++ b/addons/survey/models/res_partner.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     certifications_count = fields.Integer('Certifications Count', compute='_compute_certifications_count')
     certifications_company_count = fields.Integer('Company Certifications Count', compute='_compute_certifications_company_count')

--- a/addons/survey/models/survey_user_input.py
+++ b/addons/survey/models/survey_user_input.py
@@ -702,6 +702,7 @@ class SurveyUser_Input(models.Model):
 
 
 class SurveyUser_InputLine(models.Model):
+    _name = 'survey.user_input.line'
     _description = 'Survey User Input Line'
     _rec_name = 'user_input_id'
     _order = 'question_sequence, id'
